src/cpp/session/modules/SessionReticulate.cpp

```diff
@@ -56,6 +56,27 @@ void onDeferredInit(bool)
 }
 
 } // end anonymous namespace
+
+bool isPythonInitialized()
```

Comment:
Is this called a lot? (if so wondering if it'd be better to detect python initialization eagerly instead of recomputing it on every call when uninitialized)

Reply:
It currently doesn't, but since we do get notified when Python is initialized by `reticulate` we can just use that to set and store a flag rather than making this a more expensive call. I've made that change.